apisix/core/ai.lua

```diff
@@ -36,8 +55,8 @@ end
 
 
 local function ai_match(ctx)
-    -- TODO: we need to generate cache key dynamically
-    local key = ctx.var.uri .. "-" .. ctx.var.method .. "-" .. ctx.var.host
+    local key = get_cache_key_func(ctx)
+    core.log.info("route cache key: ", core.log.delay_exec(encode_base64, key))
```

Comment:
@kingluo after my benchmark, we have to remove this line.

so we have to find a new way for testing.